provider/pkg/gen/dotnet-templates/yaml/yaml.tmpl

```diff
@@ -161,7 +161,8 @@ namespace Pulumi.Kubernetes.Yaml
 
         }
 
-        internal static bool IsUrl(string s) => s.StartsWith("http://") || s.StartsWith("https://");
+        internal static bool IsUrl(string s)
+            => s.StartsWith("http://", StringComparison.Ordinal) || s.StartsWith("https://", StringComparison.Ordinal);
```

Comment:
`StartsWith` does a culture-sensitive string comparison by default. Which means the behavior may vary depending on the system's current culture settings. In this case, we actually want an ordinal comparison (literally comparing the bytes of the string), which will also be faster. https://docs.microsoft.com/en-us/dotnet/standard/base-types/best-practices-strings